src/api/app/helpers/webui/notification_helper.rb

```diff
@@ -16,6 +16,81 @@ def filter_notification_link(link_text, amount, filter_item)
     end
   end
 
+  def notification_icon(notifiable_type)
+    return image_tag('request_icon', size: 18) if notifiable_type == 'BsRequest'
+
+    capture do
+      tag.i(class: 'fas fa-comments mr-1')
+    end
+  end
+
+  def request_state(notification)
+    state = if notification.notifiable_type == 'BsRequest'
+              notification.notifiable.state
+            elsif notification.notifiable_type == 'Review'
+              review_badge_text(notification.notifiable)
+            end
+
+    color = request_badge_color(state)
+
+    capture do
+      tag.span(state, class: "mx-1 badge badge-#{color}")
+    end
+  end
+
+  def request_badge_color(state)
+    case state
+    when :review, :new
+      'secondary'
+    when :declined, :revoke
+      'danger'
+    when :superseded
+      'warning'
+    when :accepted
+      'success'
+    when /requires review/
+      'dark'
+    end
```

Comment:
no default?

Reply:
Ok, added an `else`clause, so any other state that would appear in black.